App store inspired changes for the schema

jeremiak opened the pull request:
@melodykramer and I have a few proposed changes (we've had discussions in #7, #8, #9, and #10) that we'd like to make `.about.yml` data a bit more useful and model the kinds of things we'll need for other dashboards and projects.

Specifically, here are the things that have changed:

0. Add an array of tags that can be applied to the project
0. Add new stages, but removes the `status` key
0. Implement a notion of "users", who are people or orgs that have forked/cloned/installed our software and are using it
0. Change the ownership model to reflect 18F as a parent organization with smaller teams inside of it

cc @mbland, @gboone and @ultrasaurus - This is ready for review and totally open for suggestions. I thought it might be easier to have concrete discussions around a PR with an implementation of the schema.

mbland commented on lib/about_yml/schema.json:
```diff
@@ -4,11 +4,11 @@
     "type": "object",
     "properties": {
-        "name": {
+        "id": {
```
The change from "name" => "id" and "full_name" => "name" is significant enough I just want to draw people's attention to it. When this change is integrated into this gem, we'll have to make sure to switch the Team API and Dashboard over to it, and then flip the switch. I'll take a task to set up a staging instance of the Team API for this purpose.

arowla replied:
You could also briefly support both variants in the gem, switch over the dependent projects to the new version, and then remove the old attributes in the next version of the gem.

mbland commented on lib/about_yml/schema.json:
```diff
@@ -17,48 +17,69 @@
+                "description": "Describes whether a project team, working group/guild, etc. owns the repo",
+                "enum": ["guild", "working-group", "internal-project", "client-project"]
+            },
+            "id": {
```
I didn't notice "id" getting copied in when we were chatting about it in real-time. Mistake? Seems out of place in "owner".

jeremiak replied:
@mbland The `id` is actually under `team`. The idea here is that a group could be a `working-group` but we'd also like to know which working group it is. It isn't `name` because I thought we agreed that we should be using `id` instead of `name`.

mbland replied:
Ah, right.

mbland commented on lib/about_yml/schema.json:
```diff
@@ -17,48 +17,69 @@
         },
+        "tags": {
+            "type": "array",
+            "description": "Tags that describe the project or aspects of thr project",
```
Typo: "thr" => "the"

mbland commented on lib/about_yml/schema.json:
```diff
@@ -111,17 +132,15 @@
     },
     "required": [
-        "name", "full_name", "stage", "team", "licenses", "owner_type", "status", "testable"
+        "name", "full_name", "stage", "team", "licenses", "owner_type", "testable"
```
Whoops, need to switch to "id" and "name" here, too.

mbland commented on lib/about_yml/schema.json:
```diff
@@ -195,6 +228,21 @@
                 }
             }
+        },
+        "contact": {
```
Note to folks: This is a format that's already been rolling out, so this is a change to catch up with current practice.

jeremiak commented:
@mbland I made your smaller changes. I guess since those lines changed they get removed from the PR comment thread. Just to confirm I:
1. Updated "thr" to be "the"
2. Updated the `required` array to reference the newly named `id` and `name` fields.

jeremiak commented:
This is a big PR and that makes it hard to talk discretely about the individual (breaking) changes. I'm going to close this PR in favor of individual PRs for each distinct change:
- #16 - Add tag collection to schema
- #17 - Add users as a collection of adopters
- #18 - Remove status and add new stages
- #19 - Model ownership at team and org levels
- #20 - Use URLs for points of contact instead of emails
